### DISCUSSION TITLE
Isolate pyo3 dependency to a crate

### DISCUSSION THREAD

**Discussion by rtsscy** [PyO3/pyo3#4793] [Questions]
Hi there! I have a rust project where i would like to cross compile to multiple targets - ios, android, wasm, python. In my workspace, I have a core/common crate where i define all the common rust classes and functions. And separately, i have a separate crates where i define the interface specific to respective targets and these crates would depend on the core crate. I am trying to implement an optional dependency of pyo3 and conditional compiling by annotating cfg_attr to pyclass and pymethods only for my custom feature flag. However, i can’t seem to work out how to add cfg_attr to the #[new] or #[pyo3(signature=..)] annotations. Anybody got any ideas?

**Comment by mejrs** [comment]
This is hard to support with how rust expands proc macros. We don't support it, see https://github.com/PyO3/pyo3/pull/2786#issuecomment-1369214439 for the reasoning. You can still have conditional complication, it just requires a bit of boilerplate.
